scripts/management/xcash-maintenance

```diff
@@ -0,0 +1,264 @@
+	"Restart all services"
+	"Get X-Cash DPoPS log"
+    "Get X-Cash Daemon log"
+    "Get X-Cash RPC Wallet log"
+	"Quit"
+)
+
+function check_height(){
+	if lsof -Pi :18281 -sTCP:LISTEN -t >/dev/null ; then
+        DATA=`$CURL GET $DAEMON_URL -d '{"jsonrpc":"2.0","id":"0","method":"get_block_count"}'`
+
+		HEIGHT=$(echo "$DATA" | grep '"count"' | awk {'print $2'} | sed s'|,||g' | tr -d $'\r')
+		STATUS=$(echo "$DATA" | grep '"status"' | awk {'print $2'} | sed s'|"||g' | tr -d $'\r')
+		
+		if [ $STATUS == "OK" ]; then
+			STATUS=${green}OK${reset}
+		else 
+			SATTUS=${red}FAIL${reset}
+		fi
+
+		echo
+		echo "Blockhain sync status: $STATUS"
+		echo "Blockchain height: $HEIGHT"
+	else
+		echo
+		echo "X-Cash Daemon service is not running...."
+		echo "Check the $XCASH_DAEMON_SRV service status"
+	fi
+}
+
+function get_node_balance(){
+	if lsof -Pi :18285 -sTCP:LISTEN -t >/dev/null ; then
+		DATA=`$CURL GET $WALLET_URL -d '{"jsonrpc":"2.0","id":"0","method":"get_balance","params":{"account_index":0,"address_indices":[0]}}'`
+		
+		ADDR=$(echo "$DATA" | grep '"address"' | awk {'print $2'} | sed s'|"||g' | sed s'|,||g' | head -1 | tr -d $'\r')
+		UNSPENT_OUTPUTS=$(echo "$DATA" | grep '"num_unspent_outputs"' | awk {'print $2'} | sed s'|"||g' | sed s'|,||g' | head -1 | tr -d $'\r')
+
+		ATOMIC_TOTAL_BALANCE=$(echo "$DATA" | grep '"balance"' | awk {'print $2'} | sed s'|,||g' | head -1 | tr -d $'\r')
+		ATOMIC_UNLOCKED_BALANCE=$(echo "$DATA" | grep '"unlocked_balance"' | awk {'print $2'} | sed s'|,||g' | head -1 | sed s"|^M||g" | tr -d $'\r')
+		TOTAL_BALANCE=$((ATOMIC_TOTAL_BALANCE/1000000))
+		UNLOCKED_BALANCE=$((ATOMIC_UNLOCKED_BALANCE/1000000))
+
+		echo
+		echo "Public address:   $ADDR"
+		echo "Unspent outputs:  $UNSPENT_OUTPUTS"
+		echo "Total balance:    $TOTAL_BALANCE XCASH"
+		echo "Unlocked balance: $UNLOCKED_BALANCE XCASH" 
+	else
+		echo
+		echo "X-Cash RCP Wallet service is not running...."
+		echo "Check the $XCASH_WALLET_SRV service status"
+	fi
+}
+
+function shared_delegates_website_statistics(){
+    if lsof -Pi :18283 -sTCP:LISTEN -t >/dev/null ; then
+        DATA=`$DPOPS_QUERY/shareddelegateswebsitegetstatistics | python3 -m json.tool`
+        RANK=$(echo "$DATA" | grep '"current_delegate_rank"' | awk {'print $2'} | sed s'|,||g' | sed s'|"||g' | tr -d $'\r')
+        TOTAL_VOTES=$(($(echo "$DATA" | grep '"total_votes"' | awk {'print $2'} | sed s'|,||g' | sed s'|"||g' | tr -d $'\r')/1000000))
+        ONLINE_PCT=$(echo "$DATA" | grep '"online_percentage"' | awk {'print $2'} | sed s'|,||g' | sed s'|"||g' | tr -d $'\r')
+        ROUNDS=$(echo "$DATA" | grep '"block_verifier_total_rounds"' | awk {'print $2'} | sed s'|,||g' | sed s'|"||g' | tr -d $'\r')
+        BLOCKS_FOUND=$(echo "$DATA" | grep '"total_blocks_found"' | awk {'print $2'} | sed s'|,||g' | sed s'|"||g' | tr -d $'\r')
+        XCASH_FOUND=$(($(echo "$DATA" | grep '"total_xcash_from_blocks_found"' | awk {'print $2'} | sed s'|,||g' | sed s'|"||g' | tr -d $'\r')/1000000))
```

Comment:
Maybe do some kind of global constant for the 100000 since you use it twice? and use it in multiple functions `readonly MYVARNAME=1000000`?

Reply:
I can do it, but I prefer to use `declare -i` for integer constants, its ok for you?

Reply:
That works for me, but do you plan on including the i and g flags as well?
or do you plan on two local variables one for each function and not including the g flag?

also because
```
declare -i a=50
a=49
echo "$a"
```

will output 49, but including the r flag will cause it to error?

its my understanding that `readonly` is essentially `declare -rg`?

I think `declare -igr` or `declare -ir` would work good, unless I am missing something here about the code or the declare attribute?

Reply:
To be honest I never checked the diferences, I always used the declare "declaration" since sounds better to me :joy: but readonly works fine too. The `-g` flag is not really necessary here since we're not modifying the value inside the function because is a read only value with the `-r` flag, but to make it more understandable I can flag them as global.